File: src/com/trapps/jvsk/java/NullExample.java
package com.trapps.jvsk.java;

/**
 * @author Akash Patra
 */
public class NullExample {

    /**
     * No Initialization
     */
    public void initNulls() {
        final String firstName = null;
        String lastName = null;
        System.out.println("Null String -> FirstName: " + firstName + ", LastName: " + lastName);
    }

    /**
     * Access Null Ref
     */
    public void accessNulls() {
        final String name = null;

        // Safe way to avoid NPE
        if (null != name) {
            final int length = name.length();
            System.out.println("Access Nulls -> Name: " + name + ", Length of Name: " + length);
        }

        // To get NPE
        final int length2 = name.length();
    }
}

File: src/com/trapps/jvsk/kotlin/NullExample.kt
package com.trapps.jvsk.kotlin

/**
 * @author Akash Patra
 */
class NullExample {

    /**
     * No Initialization
     */
    fun initNulls() {
        val firstName: String? = null
        var lastName: String?
        lastName = null
        println("Allow Nulls -> FirstName: $firstName , LastName: $lastName")
    }

    /**
     * Access Null Ref
     */
    fun accessNulls() {
        val name: String? = null

        // Safe way to avoid NPE
        val length = name?.length

        println("Access Nulls -> Name: $name , Length of Name: $length")

        // To get NPE
        val length2 = name!!.length
    }
}